panther_hardware_interfaces/src/panther_system.cpp

```diff
@@ -0,0 +1,516 @@
+    bool finished_updates = motors_controller_->UpdateDriversState();
+
+    if (finished_updates) {
+      panther_system_ros_interface_.UpdateMsgDriversParameters(
+        motors_controller_->GetFrontData().GetDriverState(),
+        motors_controller_->GetRearData().GetDriverState());
+    }
+
+    roboteq_error_filter_->UpdateError(errors_filter_ids_.read_sdo, false);
+
+  } catch (const std::runtime_error & e) {
+    RCLCPP_ERROR_STREAM(logger_, "Error when trying to read drivers feedback: " << e.what());
+    roboteq_error_filter_->UpdateError(errors_filter_ids_.read_sdo, true);
+  }
+}
+
+void PantherSystem::UpdateSystemFeedback()
+{
+  try {
+    motors_controller_->UpdateSystemFeedback();
+    UpdateHwStates();
+    panther_system_ros_interface_.UpdateMsgErrorFlags(
+      motors_controller_->GetFrontData(), motors_controller_->GetRearData());
+
+    if (
+      motors_controller_->GetFrontData().IsFlagError() ||
+      motors_controller_->GetRearData().IsFlagError()) {
+      RCLCPP_ERROR_STREAM_THROTTLE(
+        logger_, steady_clock_, 5000,
+        "Error state on one of the drivers"
+          << "\nFront: " << motors_controller_->GetFrontData().GetFlagErrorLog()
+          << "\nRear: " << motors_controller_->GetRearData().GetFlagErrorLog());
+      roboteq_error_filter_->UpdateError(errors_filter_ids_.roboteq_driver, true);
+    } else {
+      roboteq_error_filter_->UpdateError(errors_filter_ids_.roboteq_driver, false);
+    }
+
+    if (
+      motors_controller_->GetFrontData().IsDataTimedOut() ||
+      motors_controller_->GetRearData().IsDataTimedOut()) {
+      RCLCPP_ERROR_STREAM(
+        logger_, "PDO data timeout: "
+                   << (motors_controller_->GetFrontData().IsDataTimedOut() ? "front " : "")
+                   << (motors_controller_->GetRearData().IsDataTimedOut() ? "rear" : ""));
+      roboteq_error_filter_->UpdateError(errors_filter_ids_.read_pdo, true);
+    } else {
+      roboteq_error_filter_->UpdateError(errors_filter_ids_.read_pdo, false);
+    }
+
+  } catch (const std::runtime_error & e) {
+    roboteq_error_filter_->UpdateError(errors_filter_ids_.read_pdo, true);
+    RCLCPP_ERROR_STREAM(logger_, "Error when trying to read feedback: " << e.what());
+  }
+}
+
+void PantherSystem::UpdateMsgErrors()
+{
+  CanErrors can_errors;
+  can_errors.error = roboteq_error_filter_->IsError();
+  can_errors.write_sdo_error = roboteq_error_filter_->IsError(errors_filter_ids_.write_sdo);
+  can_errors.read_sdo_error = roboteq_error_filter_->IsError(errors_filter_ids_.read_sdo);
+  can_errors.read_pdo_error = roboteq_error_filter_->IsError(errors_filter_ids_.read_pdo);
+  can_errors.front_data_timed_out = motors_controller_->GetFrontData().IsDataTimedOut();
+  can_errors.rear_data_timed_out = motors_controller_->GetRearData().IsDataTimedOut();
+  can_errors.front_can_net_err = motors_controller_->GetFrontData().IsCanNetErr();
+  can_errors.rear_can_net_err = motors_controller_->GetRearData().IsCanNetErr();
```

Comment:
```suggestion
  const CanErrors can_errors = {
    roboteq_error_filter_->IsError(),
    roboteq_error_filter_->IsError(errors_filter_ids_.write_sdo),
    roboteq_error_filter_->IsError(errors_filter_ids_.read_sdo),
    roboteq_error_filter_->IsError(errors_filter_ids_.read_pdo),
    motors_controller_->GetFrontData().IsDataTimedOut(),
    motors_controller_->GetRearData().IsDataTimedOut(),
    motors_controller_->GetFrontData().IsCanNetErr(),
    motors_controller_->GetRearData().IsCanNetErr()
  };
```
Follow up to the immutable CanError structure

Reply:
as above, would prefer to leave it mutable, and I think that assigns by name are more readable and it is easier to spot a mistake